helix-term/src/ui/mod.rs

```diff
@@ -506,18 +508,28 @@ pub mod completers {
             }) // TODO: unwrap or skip
             .filter(|path| !path.is_empty());
 
+        let style_from_file = |file: Cow<'_, str>| {
+            PathBuf::from_str(file.as_ref())
+                .ok()
+                .filter(|path| path.is_dir())
```

Comment:
From what I understand this will trigger a syscall per entry. Directories already end with a slash/directory separator so you can just check the raw string for it

Reply:
Thank you, I didn't know that! I've updated so it checks if the string ends with a slash instead